src/compiler/parser-decl-var.bas

```diff
@@ -854,6 +854,49 @@ private function hBuildFakeByrefInitExpr( byval dtype as integer, byval subtype
 	function = astNewDEREF( expr )
 end function
 
+'' resolve reference to a reference
+private function hResolveRefToRefInitializer( byval dtype as integer, byval expr as ASTNODE ptr ) as ASTNODE ptr
+
+	dim tree as ASTNODE ptr = any
+	dim n as ASTNODE ptr = any
+
+	'' if initializer expr is a REF VAR, then crawl the INITREE of the var
+	'' it references for a VAR initializer.  If none found just return the 
+	'' original expr.
+
+	if( expr andalso astIsDEREF( expr ) ) then
+		if( expr->l andalso astIsVAR( expr->l ) ) then
+			if( expr->l->sym andalso symbIsRef( expr->l->sym ) ) then
```

Comment:
It could be helpful to also check `symbIsVar`, since the `symbIsRef` flag can apply to other symbol types too, i.e. procedures for "return byref", and looking at `astNewVAR()` it refers to the procedure symbol in case of function pointers.

Reply:
Added symbIsVar check.  Thanks.  I think for the correct sym.